bolt/spark/array.py

```diff
@@ -98,13 +104,13 @@ def map(self, func, axes=(0,), dtype=None):
         if dtype is None:
             dtype = self._dtype
 
-        axes = sorted(axes)
-        swapped = self._configure_key_axes(axes)
+        _, value_shape = get_kv_shape(self.shape, axes)
+        swapped = self._configure_axes(axes)
 
         # Try to compute the size of each mapped element by applying func to a random array
         element_shape = None
         try:
-            element_shape = func(random.randn(*[swapped._shape[axis] for axis in axes])).shape
+            element_shape = func(random.randn(*value_shape)).shape
```

Comment:
This can and should be replaced with `swapped.values.shape`, that avoids the extra util call and leverages the consistency of our design. The shape of the values should be correct after the swap.